File: java-usaco/src/Ride.java
import java.util.Scanner;

/**
 * ID: dhriti.1
 * LANG: Java
 * PROG: ride
 */


public class Ride {

    public static int letterToNumber(char letter) {
        return letter - ((int) 'A') + 1;
    }

    public static int multiplication(String str) {
        if (str.length() == 0) return 1;
        return letterToNumber(str.charAt(0)) * multiplication(str.substring(1));
    }

    public static String ride(String comet, String group) {
        if (multiplication(comet) % 47 == multiplication(group) % 47) return "GO";
        return "STAY";
    }

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        String comet = sc.nextLine();
        String group = sc.nextLine();
        System.out.println(ride(comet, group));
        sc.close();
    }
}

File: py-usaco/src/ride.py
"""
ID: dhriti.1
LANG: PYTHON3
PROG: ride
"""


def lettertonumber(letter):
    return ord(letter) - 64


def multiplication(string):
    if len(string) == 0:
        return 1
    else:
        return lettertonumber(string[0]) * multiplication(string[1:])


def ride(comet, group):
    if multiplication(comet) % 47 == multiplication(group) % 47:
        return "GO\n"
    else:
        return "STAY\n"

if __name__ == "__main__":
    with open("ride.in", "r") as f:
        comet = f.readline().strip()
        group = f.readline().strip()
    f = open("ride.out", "w")
    f.write(ride(comet, group))